qiskit/transpiler/preset_passmanagers/level1.py

```diff
@@ -104,14 +107,66 @@ def level_1_pass_manager(pass_manager_config: PassManagerConfig) -> PassManager:
     # 1. Use trivial layout if no layout given
     _given_layout = SetLayout(initial_layout)
 
-    _choose_layout_and_score = [
-        TrivialLayout(coupling_map),
-        Layout2qDistance(coupling_map, property_name="trivial_layout_score"),
-    ]
-
     def _choose_layout_condition(property_set):
         return not property_set["layout"]
 
+    def _trivial_not_perfect(property_set):
+        # Verify that a trivial layout is perfect. If trivial_layout_score > 0
+        # the layout is not perfect. The layout is unconditionally set by trivial
+        # layout so we need to clear it before contuing.
+        if property_set["trivial_layout_score"] is not None:
+            if property_set["trivial_layout_score"] != 0:
+                return True
+            else:
+                warnings.warn(
+                    "The current implicit default of using a trivial layout if it's a "
+                    "perfect layout will change in a future release, if you're "
+                    "depending on this behavior it is better to explicit set "
+                    "layout_method='trivial' when calling transpile()",
+                    FutureWarning,
+                )
+                # stack_level is not set because targeting transpile() is difficult
+        return False
+
+    def _vf2_match_not_found(property_set):
+        # If a layout hasn't been set by the time we run vf2 layout we need to
+        # run layout
+        if property_set["layout"] is None:
+            return True
+        # if VF2 layout stopped for any reason other than solution found we need
+        # to run layout since VF2 didn't converge.
+        if (
+            property_set["VF2Layout_stop_reason"] is not None
+            and property_set["VF2Layout_stop_reason"].value != "solution found"
+        ):
```

Comment:
I think this field is an `enum` now, right?  In that case, it could just be `if property_set["VF2Layout_stop_reason"] is not VF2LayoutStopReason.SOLUTION_FOUND`, as the `None` and `.value` bits can be skipped. (etc in other files)

Reply:
Yeah, it's an enum now. I just did it this way because I was lazy :)

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7213/commits/f77e102ce962246867495b7be814e9fdf48523bc